Mono.Addins/Mono.Addins/ExtensionNode.cs

```diff
@@ -459,8 +461,16 @@ AddinLocalizer GetAddinLocalizer ()
 				return Addin.Localizer;
 
 			Addin foundAddin = addinEngine.Registry.GetAddin (addinId);
-			if (foundAddin == null || foundAddin.Description.Localizer != null)
+			ExtensionNodeDescription localizerDescription = foundAddin != null ? foundAddin.Description.Localizer : null;
+			if (foundAddin == null || localizerDescription != null) {
+				// Try checking if there's any id for a reusable localizer without having to load the addin.
+				var localizerType = addinEngine.GetLocalizerType (foundAddin.Description.Localizer);
+				if (localizerType != null) {
+					return localizerType;
```

Comment:
This value should be somehow cached.

Reply:
This is the scenario where someone registered a localizerType already. So GetLocalizerType has a fastpath for an already registered localizer

Reply:
Oh, hm, or do you mean assigning it to the runtimeaddin's localizer?

Reply:
Currently, we cache the localizer type in `AddinEngine.localizerTypes`. We could try to cache the addinlocalizer _in_ the ExtensionNode, but I'm wary of adding another field to ExtensionNode, which is used pretty often (might impact memory use).

I can do that if you want. Not sure if we should though.